File: sources/sub-projects/myNewRtapProject/myNewRtapLibrary/src/jvx_myNewProject.cpp
#include "jvx_myNewProject.h"

jvx_myNewProject::jvx_myNewProject()
{
	this->myVolume = 1.0;
}

jvx_myNewProject::~jvx_myNewProject()
{
}

jvxDspBaseErrorType jvx_myNewProject::process(jvxData** inputs, jvxData** outputs, jvxSize numChannelsIn, jvxSize numChannelsOut, jvxSize bufferSize)
{
	if ( (numChannelsIn > 0) &&	(numChannelsOut > 0) )
	{
			for (jvxSize j = 0; j < bufferSize; j++)
			{
				for (jvxSize channelIdxOut = 0; channelIdxOut < numChannelsOut; channelIdxOut++)
				{
					jvxSize channelIdxIn = channelIdxOut % numChannelsIn;
					outputs[channelIdxOut][j] = inputs[channelIdxIn][j] * this->myVolume;
				}
			}
	}
	return(JVX_DSP_NO_ERROR);
}

File: sources/sub-projects/myNewRtapProject/myNewRtapLibrary-matlab/+jvx_myNewRtapLibrary/jvx_myNewProject.m
classdef jvx_myNewProject < handle
    %jvx_myNewProject Algorithm library template
    %   provides process method, which can be called by the audio node
    
    properties
        myVolume;
    end
    
    methods
        function this = jvx_myNewProject
            this.myVolume = 1.0;
        end
        
		function outputs = process(this, inputs, numChannelsIn, numChannelsOut, bSize)            
			assert(all(size(inputs) == [numChannelsIn, bSize]), 'Incorrect block size or number of input channels');
			outputs = zeros(numChannelsOut, bSize);
	
			for channelIdxOut = 1:numChannelsOut
				channelIdxIn = mod(channelIdxOut-1, numChannelsIn) + 1;
				outputs(channelIdxOut,:) = inputs(channelIdxIn,:) * this.myVolume;
			end
	
		end
    end
    
end

